pkg/printers/internalversion/printers.go

```diff
@@ -955,6 +956,9 @@ func printPod(pod *api.Pod, options printers.GenerateOptions) ([]metav1.TableRow
 		reason = "Unknown"
 	} else if pod.DeletionTimestamp != nil {
```

Comment:
Instead of setting the reason to "Terminating" and then changing it back, what about simplifying the if condition to only evaluate to true when the reason should be set to "Terminating", something like this?

```
	} else if pod.DeletionTimestamp != nil && pod.Status.Phase == api.PodRunning {
		reason = "Terminating"
	}
```

Reply:
Already does it here I think:
https://github.com/kubernetes/kubernetes/blob/028b457cb5b5e536f8dc4cdc50bc5a90af294a98/pkg/printers/internalversion/printers.go#L868-L871

At this point in the function, we are overriding the default reason, so there should be no need to set it again.

Reply:
> Already does it here I think:
> 
> https://github.com/kubernetes/kubernetes/blob/028b457cb5b5e536f8dc4cdc50bc5a90af294a98/pkg/printers/internalversion/printers.go#L868-L871
> 
> At this point in the function, we are overriding the default reason, so there should be no need to set it again.

Ahh missed that, sorry you are right 🤦🏻

Reply:
> Instead of setting the reason to "Terminating" and then changing it back, what about simplifying the if condition to only evaluate to true when the reason should be set to "Terminating", something like this?
> 
> ```
> 	} else if pod.DeletionTimestamp != nil && pod.Status.Phase == api.PodRunning {
> 		reason = "Terminating"
> 	}
> ```

I try to change to this and my test assert that pod with `Succeed` phase and a `Completed` container reason didn't show `Succeed`
<img width="1100" alt="image" src="https://github.com/kubernetes/kubernetes/assets/21003791/6c653ed2-caaf-40bf-9ed3-3e907f315236">